tests/functional/test_copy_config_file.py

```diff
@@ -0,0 +1,87 @@
+# THIS FILE IS PART OF THE ROSE-CYLC PLUGIN FOR THE CYLC SUITE ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Functional tests for top-level function cylc.rose.entry points
+copy_config_file.
+"""
+
+import pytest
+
+from pathlib import Path
+
+from cylc.rose.entry_points import copy_config_file
+
+
+@pytest.mark.parametrize(
+    'sources, inputs, expect',
+    [
+        (
+            # Given no input returns False
+            {},
+            {'dir_': None, 'dest_root': None},
+            False
+        ),
+        (
+            # Given no input returns False
+            {},
+            {'dir_': None, 'dest_root': 'hello'},
+            False
+        ),
+        (
+            # Valid sourcedir, no rose file:
+            {'src/hello.txt': 'Hello World!'},
+            {'dir_': 'src', 'dest_root': 'hello'},
+            False
+        ),
+        (
+            # Valid sourcedir with rose file, no rose file at dest:
+            {
+                'src/rose-suite.conf': '[env]\nFOO=1',
+                'dest/jelly.txt': 'any-old-stuff'
+            },
+            {'dir_': 'src', 'dest_root': 'dest'},
+            True
+        ),
+        (
+            # Valid sourcedir with rose file, rose file at dest:
+            {
+                'src/rose-suite.conf': '[env]\nFOO=2',
+                'dest/rose-suite.conf': '[env]\nFOO=1'
+            },
+            {'dir_': 'src', 'dest_root': 'dest'},
+            True
```

Comment:
I think this is the only test we need here, just to make sure that the new contents are copied over the old.

Reply:
Killed the rest.